src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSparkIntegrationTest.java

```diff
@@ -3,22 +3,50 @@
 import org.broadinstitute.hellbender.CommandLineProgramTest;
 import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
 import org.broadinstitute.hellbender.exceptions.UserException;
-import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
-import org.broadinstitute.hellbender.utils.test.SamAssertionUtils;
 import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.SamAssertionUtils;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.io.File;
 import java.io.IOException;
-import java.util.Arrays;
 
 public final class PrintReadsSparkIntegrationTest extends CommandLineProgramTest {
 
+    private static final File TEST_DATA_DIR = getTestDataDir();
+
     @Override
     public String getTestedClassName() {
         return PrintReadsSpark.class.getSimpleName();
     }
 
+    @DataProvider(name="testingDataCRAM")
+    public Object[][] testingDataCRAM() {
+        return new String[][]{
+                {"print_reads.sorted.cram", ".sam", "print_reads.fasta"},
+                {"print_reads.sorted.cram", ".bam", "print_reads.fasta"},
+                {"print_reads.sorted.cram", ".cram", "print_reads.fasta"},
+                {"print_reads.sorted.sam", ".cram", "print_reads.fasta"},
+                {"print_reads.sorted.bam", ".cram", "print_reads.fasta"}
```

Comment:
done, for spark and non-spark tools. found a bug too: https://github.com/broadinstitute/gatk/issues/1271